recipe/meta.yaml

```diff
@@ -57,15 +58,15 @@ requirements:
     - arm-variant * {{ arm_variant_type }}  # [aarch64]
   host:
     - patchelf >=0.12  # [linux]
-    - cuda-version {{ cuda_major }}.0
+    - cuda-version {{ cuda_compiler_version }}
     - libcublas  # [(cuda_compiler_version or "").startswith("12")]
   run:
-    - {{ pin_compatible("cuda-version", max_pin="x") }}
-    - cudatoolkit >=11.1,<12  # [(cuda_compiler_version or "").startswith("11")]
-    - libcublas               # [(cuda_compiler_version or "").startswith("12")]
+    - {{ pin_compatible("cuda-version", max_pin="x", min_pin="x") }}
+    - cudatoolkit  # [(cuda_compiler_version or "").startswith("11")]
+    - libcublas    # [(cuda_compiler_version or "").startswith("12")]
   run_constrained:
     - arm-variant * {{ arm_variant_type }}  # [aarch64]
-    - {{ stdlib('c') }}
+    - __glibc >=2.17,<3.0.a0
```

Comment:
The stdlib package already exports a dependency on __glibc when stdlib is above baseline (the first version offered on the channel for the architecture).

https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fnoarch%2Fsysroot_linux-64-2.17-h4a8ded7_18.conda

Reply:
Could it be that the `run_constrained` constraint that gets generated on sysroot_linux is overruling the more permissive constraint in the `run` section:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=999334&view=logs&j=4f922444-fdfe-5dcf-b824-02f86439ef14&t=b2a8456a-fb11-5506-ca32-5ccd32538dc0&l=691

Reply:
In that case, we should be able to remove the run_constrained constraint completely here, right?

Reply:
I checked staged recipes for [the original version of this recipe](https://github.com/conda-forge/staged-recipes/pull/13576/files). It seems that it was merged before the `{{ stdlib('c') }}` template existed which is why it needed to manually constrain the `__glibc` package. I think this recipe was just incorrectly transitioned to use `{{ stdlib('c') }}` It should not have any explicit constraints on `__glibc` and just use the `{{ stdlibc('c') }}` package in the build requirements. I'm going to open a new PR to fix this and other open issues.